Upgrade to Node v4

IgorMinar opened the pull request.

IgorMinar committed:
build(travis): bump node version to 4.1.1 and npm to 2.14.5

IgorMinar committed:
build(gulp): bump check-environment node version to 4.1.1 and npm to 2.14.5

IgorMinar committed:
build(npm): update chokidar and karma* dependencies

These dependencies required upgrade in order for us to run on Node v4.x

chokidar 1.0.1 -> 1.1.0
karma 0.12.23 -> 0.13.10
karma-chorome-launcher 0.1.8 -> 0.2.0
karma-cli 0.0.4 -> 0.1.0
karma-dart 0.2.8 -> 0.3.0
karma-sauce-launcher 0.2.11 -> 0.2.14

IgorMinar committed:
build(npm): upgrade gulp-sass to v2.0.4

IgorMinar committed:
fix(tslint): fix d.ts file paths for node_modules dependencies

IgorMinar commented:
@alexeagle I fixed the build failures, so maybe the build is going to pass now. fingers crossed

@jelbourn I'm upgrading sass dependency because the old one doesn't work with node 4. are you ok with this (material is the only module using sass so I can't estimate the impact)

jelbourn commented:
SGTM. As long as the tests pass everything should be fine.

IgorMinar committed:
build(npm): resolve shrinkwrap conflicts after rebase

IgorMinar committed:
fix(karma-dart-evalcache): make the code compatible with karma 0.13.x

IgorMinar committed:
test(tools): temporarily disable tests that depend on mock-fs

mock-fs is currently incompatible with node 4.x, but a fix is in progress
https://github.com/tschaub/mock-fs/issues/59

Since we are currently not actively developing the affected broccoli plugins,
the risk of disabling these tests is low, especially in the light of
improvements we get from node 4.x.

IgorMinar committed:
style(gulp): make code more readable

IgorMinar committed:
build(travis): add travis log folding for verbose tasks

IgorMinar committed:
fix(gulp): use the new karma.Server api

mzgol commented on .travis.yml:
```diff
@@ -1,7 +1,7 @@
 sudo: false
 node_js:
-- '0.12'
+- '4.1.1'
```
Do you really want to pin it that much? 0.12 was a range (currently matching 0.12.7).

Related: https://github.com/angular/protractor/pull/2543

IgorMinar replied:
I just want to ensure that ci uses the same version as what we use locally. So my preference is to pin it down to an exact version and update that once in a while. (especially since newer node version could update npm and cause conflicts...)

mzgol replied:
Hmm, makes sense, I guess. We can still use a newer version locally but CI might be more stable when no unplanned Node update happens.

IgorMinar replied:
yup

On Wed, Sep 30, 2015 at 3:18 PM Michał Gołębiowski <notifications@github.com>
wrote:

> In .travis.yml
> <https://github.com/angular/angular/pull/4375#discussion_r40859889>:
>
> > @@ -1,7 +1,7 @@
> >  language: node_js
> >  sudo: false
> >  node_js:
> > -- '0.12'
> > +- '4.1.1'
>
> Hmm, makes sense, I guess. We can still use a newer version locally but CI
> might be more stable when no unplanned Node update happens.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/angular/pull/4375/files#r40859889>.
>

mzgol commented on .travis.yml:
```diff
@@ -58,7 +58,7 @@ before_install:
 install:
   # Update npm
-  - npm install -g npm@2.9.1
+  - npm install -g npm@2.14.4
```
npm@2.14.4 has a broken node-gyp, that's why npm is patched in Node 4.1.1 but if you install it this way you're overwrite the patch.

Current stable npm@2 is 2.14.5 that has a correct node-gyp; I'd update npm to at least that & require this version to be safe.

IgorMinar replied:
good observation! thanks for pointing it out. I'll update the version

IgorMinar committed:
fix(karma): socket.io 1.x transport is now called 'polling' instead of 'xhr-polling'

And we really need it only when using sauce, so I'm making the condition more strict.

IgorMinar commented:
all karma problems are now resolved thanks to @zzo!! Merging!

mary-poppins commented:
Merging PR #4375 on behalf of @IgorMinar to branch [presubmit-IgorMinar-pr-4375](https://github.com/angular/angular/tree/presubmit-IgorMinar-pr-4375).

mary-poppins commented:
Merging PR #4375 on behalf of @IgorMinar to branch [presubmit-IgorMinar-pr-4375](https://github.com/angular/angular/tree/presubmit-IgorMinar-pr-4375).
